### PR TITLE
Fix URLs for glib

### PR DESCRIPTION
**Pull request by heerener**
glib fetching fails:
```
==> [2022-11-15-16:58:55.103075] Checking existence of http://ftp.gnome.org/pub/gnome/sources/glib/2.74/glib-2.74.1.tar.xz
==> [2022-11-15-16:58:55.493130] Failure reading URL: Download failed: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.
c:997)> 
```
This PR fixes the URLs

### CODE REVIEW

File: var/spack/repos/builtin/packages/glib/package.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ class Glib(Package):
     """
 
     homepage = "https://developer.gnome.org/glib/"
-    url = "https://ftp.gnome.org/pub/gnome/sources/glib/2.53/glib-2.53.1.tar.xz"
+    url = "https://download.gnome.org/sources/glib/2.53/glib-2.53.1.tar.xz"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/glib/package.py:22]: FWIW. Technically this property isn't needed when `url_for_version` is implemented.

**Review by tldahlgren — APPROVED** [review]
Confirmed existing sha256 are still valid.

**Comment by tldahlgren** [comment]
Deferring merge decision to give @michaelkuhn a chance to comment.

**Review by michaelkuhn — APPROVED** [review]
Weird, fetching still works for me. I wonder why it tries to check a certificate for an HTTP URL in your case. Anyway, can't hurt to update it. Thanks!

**Review by bernhardkaindl — APPROVED** [review]
Agreement with mkuhn, should not hurt to update. Some gnome.org seem to be offline atm. Testing fetch and build with the new URL.

**Comment by haampie** [comment]
Other PRs are failing, so I think we need this indeed.

**Comment by bernhardkaindl** [comment]
> Other PRs are failing, so I think we need this indeed.

gnome.org appears to have some web site problems, I could not reach https://wiki.gnome.org/Projects/GLib before.

I hope updating the URL may help too, here is a failure in gitlab-ci: https://gitlab.spack.io/spack/spack/-/jobs/4461884:
```py
==> Installing glib-2.74.1-6x2fzofdwv2haylvjqkwpzf55zwn2ge7
==> Error: FetchError: All fetchers failed for spack-stage-glib-2.74.1-6x2fzofdwv2haylvjqkwpzf55zwn2ge7
```
@scottwittenburg I ping you from this comment to make you aware that we've some download issues from gnome.org,
maybe this URL update helps.

**Pull request merged**